### PR TITLE
[SYCL][NFC] Remove workaround for SPIR-V translator issues

### PR DESCRIPTION
**Pull request by bader**
Latest pulldown from SPIR-V translator enabled regular execution of
several LIT tests.

### CODE REVIEW

File: sycl/test/basic_tests/boolean.cpp [hunk 1]
```diff
@@ -1,7 +1,4 @@
-// TODO: Enable compilation w/o -fno-sycl-early-optimizations option.
-// See https://github.com/intel/llvm/issues/2264 for more details.
-
-// RUN: %clangxx -fsycl -fno-sycl-early-optimizations -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
 // RUN: env SYCL_DEVICE_TYPE=HOST %t.out
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
```
Comment (bader) [inline sycl/test/basic_tests/boolean.cpp:4]:
```suggestion
// RUNx: %GPU_RUN_PLACEHOLDER %t.out
```
Let's check if this test fails on GPU only...

**Review by AGindinson — APPROVED** [review]

File: sycl/test/fpga_tests/fpga_lsu.cpp [hunk 2]
```diff
@@ -1,7 +1,4 @@
-// TODO: Enable compilation w/o -fno-sycl-early-optimizations option.
-// See https://github.com/intel/llvm/issues/2264 for more details.
-
-// RUN: %clangxx -fsycl -fno-sycl-early-optimizations %s -o %t.out
+// RUN: %clangxx -fsycl %s -o %t.out
```
Comment (mlychkov) [inline sycl/test/fpga_tests/fpga_lsu.cpp:1]: I would add also test of this sample w/o optimizations to check that all works fine in both cases.

Comment (bader) [reply sycl/test/fpga_tests/fpga_lsu.cpp:1]: Is there any reason to do this for fpga_lsu test only? 
If no, I suggest file a separate request to enable such testing for all lit tests.

Comment (mlychkov) [reply sycl/test/fpga_tests/fpga_lsu.cpp:1]: I think that this testing might be added for other lit tests as well.

**Review by mlychkov — APPROVED** [review]

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**